io/ObjReader.cpp

```diff
@@ -155,17 +159,34 @@ PointId ObjReader::addPoint(PointViewPtr view, VTN vertex)
 
 void ObjReader::newVertex(double x, double y, double z)
 {
-    m_vertices.push_back({x, y, z});
+    // w defaults to 1 according to https://en.wikipedia.org/wiki/Wavefront_.obj_file
+    m_vertices.push_back({x, y, z, 1});
+}
+
+void ObjReader::newVertex(double x, double y, double z, double w)
+{
+    m_vertices.push_back({x, y, z, w});
+}
+
+// undefined texture values default to 0 according to https://en.wikipedia.org/wiki/Wavefront_.obj_file
+void ObjReader::newTextureVertex(double x)
+{
+    m_textureVertices.push_back({x, 0, 0, 0});
+}
+
+void ObjReader::newTextureVertex(double x, double y)
+{
+    m_textureVertices.push_back({x, y, 0, 0});
 }
 
 void ObjReader::newTextureVertex(double x, double y, double z)
 {
-    m_textureVertices.push_back({x, y, z});
+    m_textureVertices.push_back({x, y, z, 0});
```

Comment:
default `w` values should probably be 1.0 instead of 0.

Reply:
This doesn't get used.

Reply:
[The Wikipedia article on obj](https://en.wikipedia.org/wiki/Wavefront_.obj_file#File_format) says it should be 0. Do you want it to be 1?

```
# List of texture coordinates, in (u, [,v ,w]) coordinates, these will vary between 0 and 1. v, w are optional and default to 0.
vt 0.500 1 [0]
```

Also, in this specific case, it is the fourth element of the texture, which isn't really defined, and is never read.